src/event-handler.js

```diff
@@ -20,7 +22,55 @@ EventHandler.prototype.logEvent = function(event) {
         }
         window['optimizely'].push(optimizelyEvent);
     }
+
+    if (optimizelyFullStackEvents.events[event.EventName]) {
+        var eventKey = event.EventName,
+            userId,
+            userAttributes = {},
+            eventTags = {};
+
+        if (window.mParticle && window.mParticle.Identity) {
+            var userIdentities = window.mParticle.Identity.getCurrentUser().getUserIdentities()['userIdentites'];
+
+            switch(forwarderSettings.userIdField) {
+                case 'customerId':
+                    userId = userIdentities["customerId"];
+                    break;
+                case 'email':
+                    userId = userIdentities["email"];
+                    break;
+                case 'mpid':
+                    userId = userIdentities["mpid"];
+                    break;
+                case 'other':
+                    userId = userIdentities["other"];
+                    break;
+                case 'other2':
+                    userId = userIdentities["other2"];
+                    break;
+                case 'other3':
+                    userId = userIdentities["other3"];
+                    break;
+                case 'other4':
+                    userId = userIdentities["other4"];
+                    break;
+                default:
+                    userId = null;
+            }
```

Comment:
quick comment to add
```suggestion
                default:
                    // this should never hit, since a user is required to select from a userId type from the userIdField dropdown
                    userId = null;
            }
```